x/mongo/driver/auth/plain_test.go

```diff
@@ -145,3 +145,48 @@ func TestPlainAuthenticator_Succeeds(t *testing.T) {
 	)
 	compareResponses(t, <-c.Written, expectedCmd, "$external")
 }
+
+func TestPlainAuthenticator_SucceedsBoolean(t *testing.T) {
+	t.Parallel()
+
+	authenticator := PlainAuthenticator{
+		Username: "user",
+		Password: "pencil",
+	}
+
+	resps := make(chan []byte, 1)
+	writeReplies(resps, bsoncore.BuildDocumentFromElements(nil,
+		bsoncore.AppendBooleanElement(nil, "ok", true),
+		bsoncore.AppendInt32Element(nil, "conversationId", 1),
+		bsoncore.AppendBinaryElement(nil, "payload", 0x00, []byte{}),
+		bsoncore.AppendBooleanElement(nil, "done", true),
+	))
+
+	desc := description.Server{
+		WireVersion: &description.VersionRange{
+			Max: 6,
+		},
+	}
+	c := &drivertest.ChannelConn{
+		Written:  make(chan []byte, 1),
+		ReadResp: resps,
+		Desc:     desc,
+	}
+
+	err := authenticator.Auth(context.Background(), &Config{Description: desc, Connection: c})
+	if err != nil {
+		t.Fatalf("expected no error but got \"%s\"", err)
+	}
+
+	if len(c.Written) != 1 {
+		t.Fatalf("expected 1 messages to be sent but had %d", len(c.Written))
+	}
+
+	payload, _ := base64.StdEncoding.DecodeString("AHVzZXIAcGVuY2ls")
```

Comment:
Optional: Not handling this error may cause confusion if anyone ever updates the test and provides an invalid base64 string. Consider handling the error to make failures more obvious for future maintainers.

E.g.
```go
payload, err := base64.StdEncoding.DecodeString("AHVzZXIAcGVuY2ls")
require.NoError(t, err, "DecodeString error")
```

Reply:
Done